src/compiler/crystal/codegen/target.cr

```diff
@@ -171,7 +171,16 @@ class Crystal::Codegen::Target
       raise Target::Error.new("Unsupported architecture for target triple: #{self}")
     end
 
-    opt_level = release ? LLVM::CodeGenOptLevel::Aggressive : LLVM::CodeGenOptLevel::None
+    opt_level = case optimization_mode
+                when Compiler::OptimizationMode::Level3
+                  LLVM::CodeGenOptLevel::Aggressive
+                when Compiler::OptimizationMode::Level2
+                  LLVM::CodeGenOptLevel::Default
+                when Compiler::OptimizationMode::Level1
+                  LLVM::CodeGenOptLevel::Less
+                else
+                  LLVM::CodeGenOptLevel::None
+                end
```

Comment:
anybody know what for this opt_level, is it options for linker?

Reply:
Roughly speaking, this level is required for the codegen passes, whereas the other one is for the optimization passes